pkg/asset/manifests/cloudproviderconfig.go

```diff
@@ -92,6 +93,15 @@ func (cpc *CloudProviderConfig) Generate(dependencies asset.Parents) error {
 		}
 
 		cm.Data[cloudProviderConfigDataKey] = openstackmanifests.CloudProviderConfig(cloud.CloudConfig)
+
+		// Get the ca-cert-bundle key if there is a value for cacert in clouds.yaml
+		if caPath := cloud.CloudConfig.CACertFile; caPath != "" {
+			caFile, err := ioutil.ReadFile(caPath)
+			if err != nil {
+				return errors.Wrap(err, "failed to read clouds.yaml ca-cert from disk")
+			}
+			cm.Data["ca-bundle.pem"] = string(caFile)
```

Comment:
wait, if the cloud provider can get the CA bundle this way, why do you need openshift/machine-config-operator#1392?  And if there is some reason you need openshift/machine-config-operator#1392 to feed the cloud provider, why would you still need this here?

Reply:
That is how we are feeding the Ca certificate to cloud-provider, as well as the other components that may need it.

Reply:
Its a little weird. So the cloud provider syncs and writes the entire content of that config file to disk in library go. However, kubelet also consumes the cloud-confg, but doesnt get the cloud.conf through library-go, it gets it through the MCO. https://github.com/openshift/machine-config-operator/blob/2bd29e5005fb6f438f1b1c512176c3f7e5cf9e47/templates/master/01-master-kubelet/_base/files/cloudconfig.yaml. So since we add a path to `ca-cert` in `cloud.conf`, it expects a file to be there on startup.